src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1754,7 +1754,28 @@ size_t MergeTreeData::clearOldPartsFromFilesystem(bool force)
     return parts_to_remove.size();
 }
 
-void MergeTreeData::clearPartsFromFilesystem(const DataPartsVector & parts_to_remove)
+void MergeTreeData::clearPartsFromFilesystem(const DataPartsVector & parts)
+{
+    clearPartsFromFilesystemImpl(parts, true, nullptr);
+}
+
+void MergeTreeData::tryClearPartsFromFilesystem(const DataPartsVector & parts, NameSet & parts_failed_to_delete)
+{
+    NameSet part_names_successeded;
+    clearPartsFromFilesystemImpl(parts, false, &part_names_successeded);
```

Comment:
Maybe try/catch here instead of inside `clearPartsFromFilesystemImpl` (and also passing a flag)?

Reply:
Really, much simpler.